datahub-web-react/src/app/entity/shared/components/styled/search/MatchingViewsLabel.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import { Button, Typography } from 'antd';
+import { useListMyViewsQuery } from '../../../../../../graphql/view.generated';
+import { DEFAULT_LIST_VIEWS_PAGE_SIZE } from '../../../../view/utils';
+import { useUserContext } from '../../../../../context/useUserContext';
+
+const MatchingViewsLabel = () => {
+    const userContext = useUserContext();
+    const selectedViewUrn = userContext?.localState?.selectedViewUrn;
+
+    const { data } = useListMyViewsQuery({
+        variables: {
+            start: 0,
+            count: DEFAULT_LIST_VIEWS_PAGE_SIZE,
+        },
+        fetchPolicy: 'cache-first',
+    });
+
+    const selectedView = selectedViewUrn
+        ? data?.listMyViews?.views.find((view) => view.urn === selectedViewUrn)
```

Comment:
Need to confirm that listMyViews will have everything... If I recall correctly we also have anoter query called listGlobalViews which may have the selected viewinside..

Reply:
So actually, yeah we also need to fetch the global / public views..

You can see where we apply similar logic on line 144 of ViewSelect.tsx:

```
    /**
     * Notice - we assume that we will find the selected View urn in the list
     * of Views retrieved for the user (private or public). If this is not the case,
     * the view will be set to undefined.
     *
     * This may become a problem if a list of public views exceeds the default pagination size of 1,000.
     */
    const foundSelectedUrn =
        (privateViews.filter((view) => view.urn === selectedUrn)?.length || 0) > 0 ||
        (publicViews.filter((view) => view.urn === selectedUrn)?.length || 0) > 0 ||
        false;
```

Reply:
Do you know the difference between these queries? I think what I have is right but want to double confirm. Here's what I see from my testing:
- `listMyViews` fetches/exposes both public and private view names
- `listGlobalViews` only is able to find public views

Reply:
let me double check this...

Reply:
okay I've just checked.

So actually the names of the method reflect it well:

- listMyViews: Returns public or private views that were _CREATED BY_ the current user. However, the problem is that a user can _apply_ a view that they did not themselves create. We mainly use listMyViews to power the settings page where a user can see all the views that they have created (own). 

- listGlobalView: Returns public views only. Doesn't matter whether the current user owns those views or not.

Because of this we need 2 queries:

listMyViews where we filter for the PERSONAL view type -> fetch the private views for the user 
listGlobalViews -> Fetch all the global views for the company. (user may have one applied)

Reply:
Ahh okay cool thanks - I updated this to include those 2 cases, let me know how it looks